### PR TITLE
docs: adding dev-docs for adding a new page to the installer

### PR DESCRIPTION
**Pull request by matthew-hagemann**
Based on my learnings from stubbing out landscape integratino in #882, I have put together some `dev-docs` outlining the broad steps required for adding a new page to the insatller, as well as adding some doc comments on some relevant dart components that did not have any.

### CODE REVIEW

File: apps/ubuntu_bootstrap/lib/app/installation_step.dart [hunk 1]
```diff
@@ -6,6 +6,19 @@ import 'package:ubuntu_provision/ubuntu_provision.dart';
...
+/// Represents each step or page in the installer workflow.
+///
+/// Each step corresponds to an enum value, which defines the page's behavior
+/// and its role in the installer. The order of the enums determines the sequence
+/// in which the steps appear in the installer.
+///
+/// New pages must be added as additional enums to integrate into the workflow.
+///
+/// Properties:
+/// - `discreteStep`: Indicates whether the step is displayed as a discrete wizard step.
```
Comment (sminez) [inline apps/ubuntu_bootstrap/lib/app/installation_step.dart:18]: we probably want to explain what being a discrete wizard step means here.

File: dev-docs/adding-a-new-page-to-the-installer.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Adding a new page to the installer
+
+## Add your page
+- Under `apps/ubuntu_bootstrap/lib/pages`, create a new directory for the page you would like to add
+ie: `apps/ubuntu_bootstrap/lib/pages/<your-feature>`
+- Create at minimum the following files:
+	- `<your-feature>_page.dart`: Defines the UI for the page you would like to add to the installer.
+	- `<your-feature>_model.dart`: Implements the provider and integrates with backend services to
```
[the remark this answers is not in the document]
Comment (sminez) [reply dev-docs/adding-a-new-page-to-the-installer.md:8]: that'll probably get out of date over time, better to describe the functionality split between the files and direct people to refer to existing pages to see what that looks like

File: dev-docs/adding-a-new-page-to-the-installer.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+## Add your page
+- Under `apps/ubuntu_bootstrap/lib/pages`, create a new directory for the page you would like to add
+ie: `apps/ubuntu_bootstrap/lib/pages/<your-feature>`
+- Create at minimum the following files:
+	- `<your-feature>_page.dart`: Defines the UI for the page you would like to add to the installer.
+	- `<your-feature>_model.dart`: Implements the provider and integrates with backend services to 
+    support your page's functionality.
+- Export your page for use in the installer by adding it to `apps/ubuntu_bootstrap/lib/pages.dart`
+- Add your page to the installation steps by adding it as an enum to `InstallationStep` in 
+`apps/ubuntu_bootstrap/lib/app/installation_step.dart`. The order of the `InstallationStep` enums
```
Comment (sminez) [inline dev-docs/adding-a-new-page-to-the-installer.md:12]: do we need to call out the weird behaviour of how page transitions can affect state here? @d-loose I still don't think I'm 100% sure I have my head round what the current behaviour is other than "here be dragons"

Comment (d-loose) [reply dev-docs/adding-a-new-page-to-the-installer.md:12]: Not exactly sure what you're referring to - do you mean the fact that pages come with a `load()` method that determines whether it's skipped or not (and additionally there's some logic in the `PageConfigService` that determines whether or not a page is shown based on the whitelabel.yaml)?

File: dev-docs/adding-a-new-page-to-the-installer.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Adding a new page to the installer
+
+## Add your page
+- Under `apps/ubuntu_bootstrap/lib/pages`, create a new directory for the page you would like to add
+ie: `apps/ubuntu_bootstrap/lib/pages/<your-feature>`
```
Comment (d-loose) [inline dev-docs/adding-a-new-page-to-the-installer.md:5]: Depending on how general the page is and whether we want to use it in `ubuntu_init` or elsewhere later on, the page could be added to `ubuntu_provision` instead

File: dev-docs/adding-a-new-page-to-the-installer.md (hunk 3 above)
Comment (sminez) [reply dev-docs/adding-a-new-page-to-the-installer.md:12]: Maybe I'm mis-remembering but I thought there was cakllbacks that fired to update state as you moved between pages and that in some places pages know (need to know?) what page is navigating to them.

Comment (matthew-hagemann) [reply dev-docs/adding-a-new-page-to-the-installer.md:12]: My understanding is the gotcha here is when you have sections with their own wizard that share a single model and state is persisted within that model between wizard steps? 

ie: `pages/storage/storage_wizard.dart`

**Review by d-loose — APPROVED** [review]

**Pull request merged**